monkey/common/agent_events/abstract_agent_event.py

```diff
@@ -29,5 +32,13 @@ class AbstractAgentEvent(InfectionMonkeyBaseModel, ABC):
     timestamp: float = Field(default_factory=time.time)
     tags: FrozenSet[str] = Field(default_factory=frozenset)
 
+    @validator("tags", each_item=True)
+    def validate_tags(cls, tag):
+        if not re.match(event_tag_regex, tag):
+            raise ValueError(
+                "Event tags can only contain alphabets, numbers, underscores, and hyphens"
+            )
+        return tag
```

Comment:
Pydantic's `constr` supports regex. Can we use that instead? https://docs.pydantic.dev/latest/usage/types/#arguments-to-constr